src/main/java/org/jenkinsci/plugins/workflow/graph/GraphLookupView.java

```diff
@@ -0,0 +1,105 @@
+package org.jenkinsci.plugins.workflow.graph;
+
+import org.kohsuke.accmod.Restricted;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+/**
+ * Interface that can be exposed by objects providing means to easily look up information about the structure of a pipeline run
+ * Usually this is scoped to a specific {@link org.jenkinsci.plugins.workflow.flow.FlowExecution}.
+ *
+ * Exists because we do not want to ourselves to only using the standard implementation in {@link StandardGraphLookupView}.
+ *
+ * <strong>Implementation note:</strong>
+ * <p>Normally this should only be used internally to power APIs, but if exposed publicly remember that {@link FlowNode}s
```

Comment:
Again recommend this be restricted unless and until there is a prepared patch using it from another plugin. Otherwise you are stuck with a possibly misdesigned API, forever.